Source/DafnyRuntime/DafnyRuntime.cs

```diff
@@ -732,8 +743,8 @@ public abstract class Sequence<T>: ISequence<T>
       return hashCode;
     }
     public override string ToString() {
-      T[] elmts = Elements;
-      if (elmts is char[]) {
+      ImmutableArray<T> elmts = ImmutableElements;
+      if (elmts.ToArray() is char[]) {
```

Comment:
This line appears to make a copy of the entire array just for the sake of doing the type test. Is there some cheaper way to do this? For example, would `elmts is ImmutableArray<char>` have the same effect? If not, is there a version of `ToArray()` with parameters that bound which elements are copied? For example, if `elmts.ToArray(from: 0, to: 0)` or something like that is possible, then at least the type test wouldn't take time proportional to the size of the array.

Reply:
I don't believe `(elmts is ImmutableArray<char>)` works (the IDE notes how the statement will never be true, as opposed to the existing path).

I can definitely check a single element though - I originally didn't because I wasn't sure if we could have mixed type arrays (e.g. `object[]` where the first element is char and the rest are not). I can make this optimization though